### PR TITLE
Fix cupyx.seterr() when `linalg` not supplied

### PR DESCRIPTION
**Pull request by takagi**
Rel #4138.

Currently, `cupyx.seterr()` unintentionally set `None` to `_config.linalg` when `linalg` argument is not supplied. This PR fixes it to leave as it is in such a case.

### CODE REVIEW

File: tests/cupyx_tests/test_cupyx.py [hunk 1]
```diff
@@ -10,9 +10,13 @@
 class TestErrState(unittest.TestCase):
 
     def test_errstate(self):
+        orig = cupyx.geterr()
         with cupyx.errstate(divide=self.divide):
             state = cupyx.geterr()
             assert state['divide'] == self.divide
+            for key in state:
+                if key != 'divide':
+                    assert state[key] == orig[key]
```
Comment (asi1024) [inline tests/cupyx_tests/test_cupyx.py:19]: How about simplifying as follows?
```suggestion
            assert state.pop('divide') == self.divide
            assert state == orig
```

Comment (takagi) [reply tests/cupyx_tests/test_cupyx.py]: Looks good. I add `orig.pop('divide')` as `orig` also has `divide` key and we need to remove it to assert.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**